hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/security/RouterSecurityManager.java

```diff
@@ -58,7 +58,8 @@ public RouterSecurityManager(Configuration conf) throws IOException {
         AuthenticationMethod.KERBEROS;
     if (authMethodConfigured.equals(authMethodToInit)) {
       this.dtSecretManager = FederationUtil.newSecretManager(conf);
-      if (this.dtSecretManager == null) {
+      if (this.dtSecretManager == null
+          || this.dtSecretManager.isRunning() == false) {
```

Comment:
I guess we should do !this.dtSecretManager.isRunning().
This may also fit in a single line.